src/zod-extensions.ts

```diff
@@ -42,6 +44,23 @@ export function extendZodWithOpenApi(zod: typeof z) {
         },
       },
     });
+
+    if (isZodType(this, 'ZodObject')) {
+      const initialExtend = this.extend;
+
+      // TODO: This does an overload everytime. So .extend().openapi() makes this change twice
+      result.extend = function (...args: any) {
+        const extendedResult = initialExtend.apply(result, args);
+
+        extendedResult._def.openapi = {
+          extendedFrom: result._def.openapi?.refId,
+        };
```

Comment:
Aren't we losing the other keys here?

Reply:
I was thinking about for some time. What I got to was something along the lines of:
1. We have an object with its metadata added into a refference
2. We create a new object with extend. This is the creation of a completely separate instance that is not meant to have the parent object's metadata since it would use `allOf` in the docs.
3. This is like a constructor so it should not be possible to have any other relevant metadata inside of it. The `refId` that it would get registered by comes after the schema is generated so they do not interfere with each other.

And yet at the same time I did have my inner doubts. I'll see if there is any scenario that I can think of, where this won't work. Feel free to help me with this, tho :smile: